code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -125,7 +125,7 @@
 				affect_ingest(M, alien, removed * ingest_abs_mult)
 			if(CHEM_TOUCH)
 				affect_touch(M, alien, removed)
-	if(overdose && (volume > overdose) && (active_metab.metabolism_class != CHEM_TOUCH && !can_overdose_touch))
+	if(overdose && (volume > overdose * M.species.chemOD_threshold) && (active_metab.metabolism_class != CHEM_TOUCH && !can_overdose_touch))

```

Comment:
What if species is null? All of these changes to reagents can runtime, because you're just given a carbon mob, without any guarantee as to the value of its variables

Reply:
Only cases of null species I can see are;
1) brains - should be fixed by having them inherit species from the mob they're spawned in, or else default to random/human
2) xeno larvae - all other xenos have their own species, so larvae should too

Reply:
My point is rather that you shouldn't assume something will be non-null because errors can occur, whether by coders or by admins messing with things they don't fully understand, so if it's possible to prevent a runtime by spotting it before the code is run, then it's generally a good idea to include some sort of check to nip the issue in the bud. In this case, I'm pretty sure just using the ?. operator on species (so, `M?.species.chemOD_threshold`) would prevent it from trying to index a variable on species if it's null, thereby preventing the runtime.

Reply:
Hadn't heard of that trick before, so I'll work it in and see what BYOND thinks.

What about the similar uses of M.species on lines 74+75 where it isn't being shunted along to the /human/ subtype?

Reply:
Those're also a risk of being runtimes, that I've never seen them runtime is in part a testament to how reliably species is non-null. Still bad practice, but since it's not directly related to your work or a procedural shift we're trying to implement on a large scale (like to_chat() used to be, or enforcing absolute pathing) I don't feel a need to demand that it be fixed.

Reply:
Fixed my instance to `M?` and left the others intact, then. 👌